src/libecalc/core/models/model_input_validation.py

```diff
@@ -0,0 +1,209 @@
+from enum import Enum
+from typing import List, Optional, Tuple
+
+import numpy as np
+from libecalc.common.logger import logger
+from numpy.typing import NDArray
+
+
+class ModelInputFailureStatus(str, Enum):
+    INVALID_RATE_INPUT = "INVALID_RATE_INPUT"
+    INVALID_SUCTION_PRESSURE_INPUT = "INVALID_SUCTION_PRESSURE_INPUT"
+    INVALID_INTERMEDIATE_PRESSURE_INPUT = "INVALID_INTERMEDIATE_PRESSURE_INPUT"
+    INVALID_DISCHARGE_PRESSURE_INPUT = "INVALID_DISCHARGE_PRESSURE_INPUT"
+    NO_FAILURE = None
+
+
+def validate_model_input(
+    rate: NDArray[np.float64],
+    suction_pressure: NDArray[np.float64],
+    discharge_pressure: NDArray[np.float64],
+    intermediate_pressure: Optional[NDArray[np.float64]] = None,
+) -> Tuple[
+    NDArray[np.float64],
+    NDArray[np.float64],
+    NDArray[np.float64],
+    NDArray[np.float64],
+    List[ModelInputFailureStatus],
+]:
+    indices_to_validate = _find_indices_to_validate(rate=rate)
+    validated_failure_status = [None] * len(suction_pressure)
+    validated_rate = rate.copy()
+    validated_suction_pressure = suction_pressure.copy()
+    validated_discharge_pressure = discharge_pressure.copy()
+    if intermediate_pressure is not None:
+        validated_intermediate_pressure = intermediate_pressure
+    if len(indices_to_validate) >= 1:
+        (
+            tmp_rate,
+            tmp_suction_pressure,
+            tmp_discharge_pressure,
+            tmp_intermediate_pressure,
+            tmp_failure_status,
+        ) = _validate_model_input(
+            rate=rate[:, indices_to_validate] if np.ndim(rate) == 2 else rate[indices_to_validate],
+            suction_pressure=suction_pressure[indices_to_validate],
+            discharge_pressure=discharge_pressure[indices_to_validate],
+            intermediate_pressure=intermediate_pressure[indices_to_validate]
+            if intermediate_pressure is not None
+            else None,
+        )
+
+        if np.ndim(rate) == 2:
+            validated_rate[:, indices_to_validate] = tmp_rate
+        else:
+            validated_rate[indices_to_validate] = tmp_rate
+        validated_suction_pressure[indices_to_validate] = tmp_suction_pressure
+        validated_discharge_pressure[indices_to_validate] = tmp_discharge_pressure
+        if intermediate_pressure is not None:
+            validated_intermediate_pressure[indices_to_validate] = tmp_intermediate_pressure
+        for i, failure in enumerate(tmp_failure_status):
+            validated_failure_status[indices_to_validate[i]] = failure
+
+    # any remaining zero or negative suction/discharge pressures (for unvalidated time steps, others are already changed)
+    # must be set to 1 (for neqsim to initiate fluid streams)
```

Comment:
Why do we have unvalidated timesteps, and why is the pressure values relevant then? Unvalidated has zero rate or negative as far as I understand. I assume it is related to multiple streams, but not sure

Reply:
I actually think it is the current implementation of simplified train that requires it. 

(https://github.com/equinor/ecalc/blob/c7f5a9955c55fdc970f60b6aade8a0793acab27a/src/libecalc/core/models/compressor/train/simplified_train.py#L162)